cpp/src/replace/replace.cu

```diff
@@ -77,20 +92,39 @@ namespace{ //anonymous
   /* ----------------------------------------------------------------------------*/
   struct replace_kernel_forwarder {
     template <typename col_type>
-    void operator()(void*       d_col_data,
-                    gdf_size_type      nrows,
-                    const void* d_old_values,
-                    const void* d_new_values,
-                    gdf_size_type      nvalues)
+    void operator()(gdf_column*       col,
+                    const gdf_column* old_values,
+                    const gdf_column* new_values)
     {
-      thrust::device_ptr<const col_type> old_values_begin = thrust::device_pointer_cast(static_cast<const col_type*>(d_old_values));
+      const bool col_is_nullable = (col->valid != nullptr);
```

Comment:
You should check that it actually has null values too, i.e., `null_count > 0`.  No need to have extra null logic if the `null_count == 0`.

Reply:
According to current implementation, there is one case when, there are nulls in` new_column` but no nulls in `col`, In this case, I am the pre-allocating the mask for `col` despite having `null_count=0`. So, `null_count>0` can not be applied on `col` but surely can be applied on `new_values`.
Now, I am redesigning the replace API, then `col->valid != nullptr && col->null_count > 0` can be applied.